k-distribution/k-tutorial/1_basic/15_configurations/README.md

```diff
@@ -1 +1,245 @@
 # Lesson 1.15: Configuration Declarations and Cell Nesting
+
+The purpose of this lesson is to explain how to store additional information
+about the state of your interpreter by declaring additional cells using the
+`configuration` sentence, as well as how to add additional inputs to your
+definition.
+
+## Cells and Configuration Declarations
+
+We have already covered the absolute basics of cells in K by looking at the
+`<k>` cell. As explained in [Lesson 1.13](../13_rewrite_ruels/README.md), the
+`<k>` cell is available without being explicitly declared. It turns out this is
+because, if the user does not explicitly specify a `configuration` sentence
+anywhere in the main module of their definition, the `configuration` sentence
+from the `DEFAULT-CONFIGURATION` module of
+[kast.md](../../../include/kframework/builtin/kast.md) is imported
+automatically. Here is what that sentence looks like:
+
+```
+  configuration <k> $PGM:K </k>
+```
+
+This configuration declaration declares a single cell, the `<k>` cell. It also
+declares that at the start of rewriting, the contents of that cell should be
+initialized with the value of the `$PGM` **configuration variable**. A
+configuration variable is like an input to `krun`. It is a term that is
+supplied to krun in the form of an AST that was parsed from the grammar of
+a particular module. By default, the `$PGM` configuration variable uses the 
+module of the main syntax module of the definition. The cast on the
+configuration variable also specifies the sort that is used as the entry point
+to the parser, in this case the `K` sort.
+
+Note that we did not explicitly specify the `$PGM` configuration variable when
+we invoke `krun` on a file. This is because `krun` specifically handles a file
+passed as a positional argument as a file containing the term that should be
+passed as the `$PGM` configuration variable, as a special naming convention.
+We did, however, specify the `PGM` name explicitly when we called `krun` with
+the `-cPGM` command line argument. This is the other, explicit, way of
+specifying an input to krun.
```

Comment:
```suggestion
Note that we did not explicitly specify the `$PGM` configuration variable when
we invoke `krun` on a file. This is because `krun` handles the `$PGM` variable
specially, and assumes that the contents of the file you are calling `krun` on
should be used for the `$PGM` configuration variable.
We did, however, specify the `PGM` name explicitly when we called `krun` with
the `-cPGM` command line argument in [Lesson 1.02](../02_basicsREADME.md).
This is the other, explicit, way of specifying an input to krun.
```